### PR TITLE
Examples: add PyVista/VTK example

### PR DESCRIPTION
**Pull request by MuellerSeb**
Closes: #93 

This PR finally adds a mesh example using PyVista. 

Plotting is mocked with static image, since, Pyvista plotting is not working on ReadTheDocs.
We would need to switch to GitHub pages in order to get that right.
See: https://github.com/pyvista/pyvista/issues/185

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -125,3 +125,4 @@ docs/source/generated/
 
 *.vtu
 *.vtr
+*.vtk
```
Comment (LSchueler) [inline .gitignore:128]: These endings are getting excessive...

File: docs/source/conf.py [hunk 2]
```diff
@@ -260,11 +261,22 @@ def setup(app):
...
+# https://github.com/simpeg/discretize/blob/f414dd7ee7c5ba9a141cb2c37d4b71fdc531eae8/docs/conf.py#L334
+# Make sure off screen is set to true when building locally
+# pyvista.OFF_SCREEN = True
+# # necessary when building the sphinx gallery
+# pyvista.BUILDING_GALLERY = True
+# # Optional - set parameters like theme or window size
+# pyvista.set_plot_theme("document")
+
 sphinx_gallery_conf = {
+    # "image_scrapers": ("pyvista", "matplotlib"),
```
Comment (LSchueler) [inline docs/source/conf.py:279]: Why is this still there?

File: examples/00_misc/03_precipitation.py [hunk 3]
```diff
@@ -153,5 +153,5 @@
 # We will cut out the volumes with low precipitation in order to have a look
 # inside the cuboid.
 
-mesh = srf.to_pyvista()
-mesh.threshold_percent(0.25).plot()
+# mesh = srf.to_pyvista()
+# mesh.threshold_percent(0.25).plot()
```
Comment (LSchueler) [inline examples/00_misc/03_precipitation.py:157]: This is gonna be outdated real soon ;-)

File: examples/01_random_field/06_pyvista_support.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#    uncommenting the following line of code.
+
+# grid.contour(isosurfaces=8).plot()
+
+###############################################################################
+# The result should look like this:
+#
+# .. image:: https://github.com/GeoStat-Framework/GeoStat-Framework.github.io/raw/master/img/GS_pyvista_cut.png
+#    :width: 400px
+#    :align: center
```
Comment (LSchueler) [inline examples/01_random_field/06_pyvista_support.py:56]: That's a good solution for now.

**Review by LSchueler — APPROVED** [review]
Nice! More tutorials!!!
I think this is ready to be merged.
But don't forget to pull, I've fixed some very few) typos.

File: docs/source/conf.py (hunk 2 above)
Comment (MuellerSeb) [reply docs/source/conf.py:279]: I wanted to keep it, when pyvista fixes this off-screen rendering bug. (So we don't have to look it up again)

File: examples/00_misc/03_precipitation.py (hunk 3 above)
Comment (MuellerSeb) [reply examples/00_misc/03_precipitation.py:157]: True that :wink:

File: examples/01_random_field/06_pyvista_support.py (hunk 4 above)
Comment (MuellerSeb) [reply examples/01_random_field/06_pyvista_support.py:56]: At least to show off (a bit).

File: .gitignore (hunk 1 above)
Comment (MuellerSeb) [reply .gitignore:128]: I know...

**Pull request merged**